pkg/spoke/controllers/manifestcontroller/manifestwork_controller.go

```diff
@@ -271,12 +293,57 @@ func (m *ManifestWorkController) applyOneManifest(
 	// TODO we should check the certain error.
 	// Use dynamic client when scheme cannot decode manifest or typed client cannot handle the object
 	if isDecodeError(result.Error) || isUnhandledError(result.Error) || isUnsupportedError(result.Error) {
-		result.Result, result.Changed, result.Error = m.applyUnstructured(ctx, unstructuredObj, gvr, recorder)
+		result.Result, result.Changed, result.Error = m.applyUnstructured(ctx, required, gvr, recorder)
 	}
 
 	return result
 }
 
+func (m *ManifestWorkController) serverSideApply(
+	ctx context.Context,
+	gvr schema.GroupVersionResource,
+	required *unstructured.Unstructured,
+	config *workapiv1.ServerSideApplyConfig,
+	recorder events.Recorder) (*unstructured.Unstructured, bool, error) {
+	force := false
+	fieldManager := "work-agent"
+
+	if config != nil {
+		force = config.Force
+		fieldManager = config.FieldManager
+	}
+
+	existing, err := m.spokeDynamicClient.
+		Resource(gvr).
+		Namespace(required.GetNamespace()).
+		Get(ctx, required.GetName(), metav1.GetOptions{})
+
+	switch {
+	case err == nil:
+		// merge ownerref by existing
+		modified := resourcemerge.BoolPtr(false)
+		existingOwners := existing.GetOwnerReferences()
+		resourcemerge.MergeOwnerRefs(modified, &existingOwners, required.GetOwnerReferences())
+		required.SetOwnerReferences(existingOwners)
+	case !errors.IsNotFound(err):
+		return nil, false, err
+	}
+
+	patch, err := json.Marshal(resourcemerge.WithCleanLabelsAndAnnotations(required))
+	if err != nil {
+		return nil, false, err
+	}
+
+	actual, err := m.spokeDynamicClient.
+		Resource(gvr).
+		Namespace(required.GetNamespace()).
+		Patch(ctx, required.GetName(), types.ApplyPatchType, patch, metav1.PatchOptions{FieldManager: fieldManager, Force: pointer.Bool(force)})
```

Comment:
Does this work?  I'd be very happy to see a k/k patch to add Apply to the dynamic client.

Reply:
Apply is in the client-go's latest code, but not in 2.4. Will add a TODO to change this to Apply